### DISCUSSION TITLE
Binding from code for InlineCollection

### DISCUSSION THREAD

**Discussion by tobyfirth** [AvaloniaUI/Avalonia#16025] [Q&A]
I have been having a problem with binding from code for a Run in an InlineCollection. Using the example for binding from code for the documentation the binding seems to be set up with the background property having the expected colour when first rendered.
```
var run = new Run();

var binding = new Binding 
{ 
    Source = someObjectImplementingINotifyPropertyChanged, 
    Path = nameof(someObjectImplementingINotifyPropertyChanged.MyProperty)
}; 

run.Bind(TextElement.BackgroundProperty, binding);
```

However, when the property that it is bound to changes, the change is not applied on the screen. The Run defiantly has the new value as I can subscribe to the background property and get the value change. Using the DevTools I can see that the background property has changed. When the width of the window is changed the background does change though (doesn't happen when changing the height). 

If I do the binding from Xaml then everything works as expected.
`<Run Background="{Binding Path.To.Property}"/>`

Presumably there is something that the Xaml generated binding has set that I am not setting which would enable this to work. I have looked through the various properties of the binding and tried various options but I have not yet managed to make it work.
What am I missing here?

**Comment by stevemonaco** [comment]
`Path = nameof(someObjectImplementingINotifyPropertyChanged.MyProperty)`

`nameof` here only returns "MyProperty", so it's not a subproperty binding. Double check that it's a valid path.

**Reply by tobyfirth** [reply]
If I do exactly the same thing using TextBlock then everything works as expected. My best guess is that when the value changes the invalidate function for some reason is not running when the property changes with my binding.
